src/core/config/createValidator.js

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { assign } from "../../utils";
+
+const CONFIG_DOC_URI =
+  "https://launch.gitbook.io/adobe-experience-platform-web-sdk/fundamentals/configuring-the-sdk";
+
+const createConfigValidator = config => {
+  const validators = Object.create(null);
+
+  const validator = {
+    addValidators(newValidators) {
+      return assign(validators, newValidators);
```

Comment:
adding isRequired creates a weird edge case when isRequired is true, and there is a defaultValue.  We could throw an error or just ignore the isRequired field as this code currently does.